lib/aasm/base.rb

```diff
@@ -63,21 +63,39 @@ def initial_state(new_initial_state=nil)
     end
 
     # define a state
-    def state(name, options={})
-      @state_machine.add_state(name, @klass, options)
-
-      if @klass.instance_methods.include?("#{name}?")
-        warn "#{@klass.name}: The aasm state name #{name} is already used!"
+    # args
+    # [0] state
+    # [1] options (or nil)
+    # or
+    # [0] state
+    # [1..] state
+    def state(*args)
+      if args.last.is_a?(Hash) && args.size == 2
+        names = [args.first]
+        options = args.last
+      elsif
+        names = args
+        options = {}
+      else
+        raise "count not parse states: #{args}"
       end
 
```

Comment:
I am wondering whether it would make more sense to have what follows as a separate (private) method which is called directly from this method with the correct arguments. 

e.g. lines 74-75 becoming `new_method_name(args.first, args.last)` etc.